File: typescript/difference-of-squares/difference-of-squares.ts
export default class Squares {
  squareOfSum: number;
  sumOfSquares: number;
  firstN: number[];
  difference: number;

  constructor(num: number) {
    this.firstN = this.createFirstN(num);
    this.squareOfSum = this.calcSquareOfSum();
    this.sumOfSquares = this.calcSumOfSquares();
    this.difference = this.squareOfSum - this.sumOfSquares;
  }

  createFirstN(num: number): number[] {
    const result: number[] = [];
    for (let i: number = 1; i <= num; i++) {
      result.push(i);
    }
    return result;
  }

  calcSquareOfSum(): number {
    return this.firstN.reduce((res, int) => (res += int), 0) ** 2;
  }

  calcSumOfSquares(): number {
    return this.firstN.reduce((res, int) => (res += int ** 2), 0);
  }
}

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SquareOfSum calculates the square of the sum of positive ints up to num.
func SquareOfSum(num int) (res int) {
	sum := (num * (num + 1)) / 2
	return sum * sum
}

// SumOfSquares calculates the sum of positive ints squared up to num.
func SumOfSquares(num int) (res int) {
	return num * (num + 1) * (2*num + 1) / 6
}

// Difference calculates the difference between SquareOfSum and SumOfSquares.
func Difference(num int) (res int) {
	return SquareOfSum(num) - SumOfSquares(num)
}
